src/OpenTelemetry/Trace/ExceptionProcessor.cs

```diff
@@ -0,0 +1,53 @@
+// <copyright file="ExceptionProcessor.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace OpenTelemetry.Trace
+{
+    internal class ExceptionProcessor : BaseProcessor<Activity>
+    {
+        private readonly Func<IntPtr> fnGetExceptionPointers;
+
+        public ExceptionProcessor()
+        {
+            try
+            {
+                var flags = BindingFlags.Static | BindingFlags.Public;
+                var method = typeof(Marshal).GetMethod("GetExceptionPointers", flags, null, new Type[] { }, null);
+                var lambda = Expression.Lambda<Func<IntPtr>>(Expression.Call(method));
+                this.fnGetExceptionPointers = lambda.Compile();
+            }
+            catch (Exception ex)
+            {
+                throw new NotSupportedException("System.Runtime.InteropServices.Marshal.GetExceptionPointers is not supported.", ex);
+            }
+        }
+
+        /// <inheritdoc />
+        public override void OnEnd(Activity activity)
+        {
+            if (this.fnGetExceptionPointers() != IntPtr.Zero)
+            {
+                activity.SetStatus(Status.Error);
```

Comment:
Is it possible to get the type, message, and/or stack from the IntPtr returned from GetExceptionPointers?

Reply:
Yes it is possible by traveling through the SEH object. This is CPU/OS/platform dependent and not well documented.
https://bytepointer.com/resources/pietrek_crash_course_depths_of_win32_seh.htm.

Reply:
Looks petty intense! Not sure if it is worth attempting.

Why I bring it up:

I was just thinking about the applications I have instrumented with OpenTelemetry .NET so far. There are really only a few spots I've had the need to create spans manually. Almost exclusively around queues/topics. In those cases, I'm doing what the example has, with one exception...
```csharp
catch (Exception ex)
{
   activity.SetStatus(Status.Error.WithDescription(ex.Message));
   throw;
}
```
...instead of...
```csharp
catch (Exception)
{
   activity.SetStatus(Status.Error);
   throw;
}
```
So, commenting from the perspective of a user of the library, since there's only a handful of spots where I have the try/catch, and the message/description is super useful, I'll probably continue doing the try/catch instead of using the processor/option. But if the processor could grab the message too, that would be something.

Reply:
I'm happy to cover it in a follow PR.
One heads up, the exceptions we captured here could be any SEH exception - e.g. C++ exception, OS exception.

Reply:
I'm using #1874 to explore how deep should we go.